libs/dh/admin/feature-permissions/src/lib/details/tabs/dh-admin-permission-audit-logs.component.ts

```diff
@@ -61,69 +66,38 @@ import { PermissionAuditLog } from '../../permissionAuditLog';
     WattDatePipe,
   ],
 })
-export class DhPermissionAuditLogsComponent implements OnInit, OnChanges, OnDestroy {
-  private trans = inject(TranslocoService);
-  dataSource = new WattTableDataSource<PermissionAuditLog>();
-
-  @Input({ required: true }) selectedPermission!: PermissionDto;
+export class DhPermissionAuditLogsComponent implements OnChanges {
+  private readonly apollo = inject(Apollo);
```

Comment:
Issue: valueChange observeble should be destoryed

```
export class DhPermissionAuditLogsComponent implements OnChanges {
  private readonly apollo = inject(Apollo);
  private readonly getPermissionAuditLogsQuery = this.apollo.watchQuery({
    useInitialLoading: true,
    notifyOnNetworkStatusChange: true,
    query: GetPermissionAuditLogsDocument,
    fetchPolicy: 'cache-and-network',
  });

  private readonly getPermissionsAuditLogs$ =
    this.getPermissionAuditLogsQuery.valueChanges.pipe(takeUntilDestroyed());

  dataSource = new WattTableDataSource<PermissionAuditedChangeAuditLogDto>();

  hasFailed$ = this.getPermissionsAuditLogs$.pipe(
    map((result) => !!result.error),
    catchError(() => of(true))
  );

  isLoading$ = this.getPermissionsAuditLogs$.pipe(
    tap(
      (result) => (this.dataSource.data = [...(result.data?.permissionAuditLogs ?? [])].reverse())
    ),
    map((result) => result.loading),
    catchError(() => of(false))
  );

  columns: WattTableColumnDef<PermissionAuditedChangeAuditLogDto> = {
    timestamp: { accessor: 'timestamp' },
    entry: { accessor: null },
  };

  @Input({ required: true }) selectedPermission!: PermissionDto;

  ngOnChanges(): void {
    this.getPermissionAuditLogsQuery?.refetch({ id: this.selectedPermission?.id });
  }
}```

Reply:
Why is it necessary? Is the entire point of letting only angular subscribe is that it can unsubscribe too?

Reply:
You are right !